### PR TITLE
 list: add getRole and isChecked to ariaProvider

### PR DESCRIPTION
**Pull request by isidorn**
This PR does the following:
* Renames lists `IAriaSetProvider` to `IAriaProvider`
* Adds `getRole` and `isChecked` optional methods to the `IAriaProvider`
* Contributes the `getRole` and `isChecked` for the breakpoints view. Such that the breakpoints are now properly annouched as checkmarks. 

I have tested this and all works good on my mac.

### CODE REVIEW

File: src/vs/base/browser/ui/list/listView.ts [hunk 1]
```diff
@@ -566,7 +568,12 @@ export class ListView<T> implements ISpliceable<T>, IDisposable {
 
 		if (!item.row) {
 			item.row = this.cache.alloc(item.templateId);
-			item.row!.domNode!.setAttribute('role', 'treeitem');
+			const role = this.ariaProvider.getRole ? this.ariaProvider.getRole(item.element) : 'treeitem';
+			item.row!.domNode!.setAttribute('role', role);
+			const checked = this.ariaProvider.isChecked ? this.ariaProvider.isChecked(item.element) : undefined;
+			if (checked !== undefined) {
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/list/listView.ts]: `typeof checked !== 'undefined'`

**Review by joaomoreno — APPROVED** [review]
Just one nitpick.

File: src/vs/base/browser/ui/list/listView.ts (hunk 1 above)
Comment (isidorn) [reply src/vs/base/browser/ui/list/listView.ts]: Ack.
I was not aware of the difference thus I found this https://stackoverflow.com/questions/4725603/variable-undefined-vs-typeof-variable-undefined

**Comment by isidorn** [comment]
Thanks for the review.

**Pull request merged**